builtin/array.mbt

```diff
@@ -835,6 +835,27 @@ pub fn dedup[T : Eq](self : Array[T]) -> Unit {
   }
 }
 
+/// Removes consecutive repeated elements in the array according to the Eq trait.
+/// 
+/// # Example
+/// ```
+/// let v = [3, 4, 5, 6, 6, 4]
```

Comment:
1. investigate [... @hash_set.from_iter(...)] should preserve the order
2. dedup() (breaking changes, document assuming the input is already sorted